executors/rust/1.3/src/datetimefmt.rs

```diff
@@ -0,0 +1,158 @@
+
+    let date_style_str = &option_struct.date_style;
+    let date_style = if date_style_str == &Some("full".to_string()) {
+        length::Date::Full
+    } else if date_style_str == &Some("long".to_string()) {
+        length::Date::Long
+    } else if date_style_str == &Some("short".to_string()) {
+        length::Date::Short
+    } else if date_style_str == &Some("medium".to_string()) {
+        length::Date::Medium
+    } else {
+        length::Date::Full
+    };
+
+    // TimeStyle long or full requires that you use ZonedDateTimeFormatter.
+    // long known issue that is documented and has been filed several times.
+    // Is fixed in 2.0
+    let time_style_str = &option_struct.time_style;
+    let time_style = if time_style_str == &Some("full".to_string()) {
+        length::Time::Full
+    } else if time_style_str == &Some("long".to_string()) {
+        length::Time::Long
+    } else if time_style_str == &Some("short".to_string()) {
+        length::Time::Short
+    } else if time_style_str == &Some("medium".to_string()) {
+        length::Time::Medium
+    } else {
+        // !!! SET TO UNDEFINED
+        length::Time::Full
+    };
+
+    // Set up DT option if either is set
+    let dt_options = if date_style_str.is_some() && time_style_str.is_some() {
+        length::Bag::from_date_time_style(date_style, time_style)
+    } else if date_style_str.is_none() && time_style_str.is_some() {
+        length::Bag::from_time_style(time_style)
+    } else if date_style_str.is_some() && time_style_str.is_none() {
+        length::Bag::from_date_style(date_style)
+    } else {
+        length::Bag::default()
+    };
+
+    // Get ISO input string
+    let input_time_string = &json_obj["input_string"].as_str().unwrap();
+    let input_iso: String = input_time_string.to_string() + "[-00:00]";
+
+    let dt_iso = IxdtfParser::new(&input_iso).parse().unwrap();
+    let date = dt_iso.date.unwrap();
+    let time = dt_iso.time.unwrap();
+    // let offset = dt_iso.offset.unwrap();
+    // let tz_annotation = dt_iso.tz;
+
+    let datetime_iso = DateTime::try_new_iso_datetime(
+        date.year,
+        date.month,
+        date.day,
+        time.hour,
+        time.minute,
+        time.second,
+    )
+    .expect("Failed to initialize ISO DateTime instance.");
+    let any_datetime = datetime_iso.to_any();
+
+    // Testing with a default timezone
+    let time_zone = CustomTimeZone::utc();
+
+    // The constructor is called
+    let dtf_result =
+        ZonedDateTimeFormatter::try_new(&data_locale, dt_options.into(), Default::default());
+
+    let datetime_formatter = match dtf_result {
+        Ok(dtf) => dtf,
+        Err(e) => {
+            return Ok(json!({
+                "label": label,
+                "error_detail": format!("{option_struct:?}"),
+                "error_type": format!("Failed to create DateTimeFormatter instance: {e:?}"),
+            }));
+        }
+    };
+
+    // TODO: Use a skeleton.
```

Comment:
Elango says: Rewrite or remove comment? Is there a way to use skeleta in date time formatter in versions < 2.0?